phoenicis-repository/src/main/java/org/phoenicis/repository/DefaultRepositoryManager.java

```diff
@@ -80,6 +80,40 @@ public void moveRepository(RepositoryLocation<? extends Repository> repositoryUr
         this.triggerRepositoryChange();
     }
 
+    @Override
+    public void updateRepositories(final List<RepositoryLocation<? extends Repository>> repositoryLocations) {
+        LOGGER.info(String.format("Updating repositories list to %s", repositoryLocations.toString()));
+
+        final Map<RepositoryLocation<? extends Repository>, Repository> copy = new HashMap<>(this.repositoryMap);
+
+        // delete the old repositories
+        copy.forEach((repositoryLocation, repository) -> {
+            this.multipleRepository.removeRepository(repository);
+
+            if (!repositoryLocations.contains(repositoryLocation)) {
+                this.repositoryMap.remove(repositoryLocation);
+
+                repository.onDelete();
+            }
+        });
+
+        // add the new repositories
+        repositoryLocations.forEach(repositoryLocation -> {
+            if (!this.repositoryMap.containsKey(repositoryLocation)) {
+                final Repository repository = repositoryLocation.createRepository(cacheDirectoryPath,
+                        localRepositoryFactory,
+                        classPathRepositoryFactory, fileUtilities);
+
+                this.repositoryMap.put(repositoryLocation, repository);
+            }
+
+            this.multipleRepository.addRepository(this.repositoryMap.get(repositoryLocation));
```

Comment:
Wouldn't this duplicate the repository in `this.multipleRepository` if it is contained in `this.repositoryMap` already (for whatever reason)?
Can we avoid this duplicate maintenance of `multipleRepository` and `repositoryMap`?

Reply:
In line `91` all repositories are removed from the `multipleRepository`. This is done to ensure that:
- the ordering of the repositories inside `multipleRepository` is correct
- no duplicates are contained inside `multipleRepository`

Reply:
I guess we can remove this redundancy, but in my opinion this should be part of a complete refactoring of the whole repository model.

Reply:
Ok.